spec.md

```diff
@@ -1195,22 +1195,24 @@ A source expression supplies a signal and can be used to drive a circuit compone
 A sink expression can be driven by another expression.
 A duplex expression is an expression that is both a source and sink.
 
-The rules for the flow of an expression are as follows:
+The rules for the flow of an expression are as follows.
 
-If the expression is a reference, we look at the kind of the circuit component:
+If the expression is an identifier, we look at the kind of the circuit component the identifier refers to:
 
 -   Nodes are sources.
 -   Wires and registers are duplex.
--   For ports, `input` ports are sources and `output` ports are sinks.
+-   For ports, `input` ports are sources and `output` ports are duplex.
```

Comment:
This may not be correct as it conflates direction with flow. An output port swaps the flow, but an input port doesn't. 

What I think this means is that the list of rules later should indicate that "if a port has output direction, then its flow is reversed" and "if a port has input direction, then its flow is not reversed."